lldb/source/Target/Statistics.cpp

```diff
@@ -192,6 +192,10 @@ TargetStats::ToJSON(Target &target,
   }
   target_metrics_json.try_emplace("sourceMapDeduceCount",
                                   m_source_map_deduce_count);
+  target_metrics_json.try_emplace("sourceRealpathAttemptCount",
```

Comment:
Should you still count this if realpath_prefixes is nullptr?  In that case you definitely didn't attempt to realpath anything.

Reply:
The `TargetStatistics` class doesn't really know about `RealpathPrefixes`. It just receives the increments. In the case you mentioned, both counts will simply never be incremented and be reported as zeros.

Are you referring to these two additional fields in the output, in the sense that they waste output size and should be omitted? The output size of `statistics dump` can already be really large esp. in the non-summary mode, so I am not sure if adding two additional counts matter that much, but if you think we should omit then I can make the counters `std::optional<uint32_t>`.

Reply:
I think I was being more simple-minded.  In 99% of the uses of lldb, this function will get a nullptr for the realpathprefixes array, and will be doing no realpath work here.  It would be good if those clients didn't suffer any penalty for this facility they aren't using.  But as it stands they going to pay the cost of sticking a 0 value into some JSON array regardless.  Why do you need to add these two fields?  They can't be required, since if I never set a file and line breakpoint in a session, I'll never enter this code, yet the target statistics still have to work.
I have no problem with adding the fields when they represent work that is being attempted.  But for the most part this is going to be work that's never attempted.

Reply:
I was missing where this work is done.  I don't care about two extra fields set once in the JSON.  This is fine.